### PR TITLE
Use Jimp for layout plugin

### LINKED ISSUE #472: Dependency `images` incompatible with ARM64 architecture
There is an incompatible dependency, `images`, which does not support ARM64 and means that we cannot build the network CDN image for this architecture.

If you need any more info, let me know.

### PR DESCRIPTION
**Pull request by jimlambie**
Jimp is already a dependency (and 100% JS) so have used this in place of the `images` dependency. This layout plugin should be removed from core in 4.0.

Close #472

### CODE REVIEW

File: workspace/plugins/layout.js [hunk 1]
```diff
@@ -78,77 +53,115 @@ ImageLayoutProcessor.prototype.get = function () {
...
+
+              let concatStream = concat(obj => {
+                return addImage(input, obj, () => {
+                  if (++i === this.inputs.length) {
+                    return returnImage(instance)
+                  }
+                })
+              })
+
+              console.log('>>>>', input.fileName, index)
```
Comment (eduardoboucas) [inline workspace/plugins/layout.js]: 🔪

File: workspace/plugins/layout.js [hunk 2]
```diff
@@ -78,77 +53,115 @@ ImageLayoutProcessor.prototype.get = function () {
...
+
+              streams[index].pipe(imageSizeStream)
+              streams[index].pipe(imageStream)
+
+              return this.getImageSize(imageSizeStream).then(imageInfo => {
+                input.originalImageSize = imageInfo
+                imageStream.pipe(concatStream)
+              })
+            } else if (input.colour) {
+              console.log('>>>>', input.colour, index)
```
Comment (eduardoboucas) [inline workspace/plugins/layout.js]: 🔫

File: workspace/plugins/layout.js [hunk 3]
```diff
@@ -78,77 +53,115 @@ ImageLayoutProcessor.prototype.get = function () {
...
             }
 
-            let extractedImage = imageLib(inputImage, input.l, input.t, input.width, input.height)
-
-            newImage.draw(extractedImage, input.x, input.y)
-          } catch (err) {
-
+            // Read the overlay image, resive and composite it on the original
+            Jimp.read(obj)
+              .then(inputImage => {
```
Comment (eduardoboucas) [inline workspace/plugins/layout.js]: Do we need a `catch`?

File: workspace/plugins/layout.js [hunk 4]
```diff
@@ -78,77 +53,115 @@ ImageLayoutProcessor.prototype.get = function () {
...
+                imageStream.pipe(concatStream)
+              })
+            } else if (input.colour) {
+              console.log('>>>>', input.colour, index)
+
+              // Create a colour tile.
+              new Jimp(input.width, input.height, `#${input.colour}`, (_err, image) => {
+                image.getBuffer(Jimp.MIME_PNG, (err, buffer) => {
+                  addImage(input, buffer, () => {
+
```
Comment (eduardoboucas) [inline workspace/plugins/layout.js]:
```suggestion
```

File: workspace/plugins/layout.js (hunk 1 above)
Comment (jimlambie) [reply workspace/plugins/layout.js]: Nope.

File: test/acceptance/visual.js [hunk 5]
```diff
@@ -75,7 +76,7 @@ function requestTestImage (test) {
 
           diff.image.write(outputPath)
 
-          return Promise.reject(error)
```
Comment (eduardoboucas) [inline test/acceptance/visual.js]: 👏

File: test/acceptance/visual.js [hunk 6]
```diff
@@ -59,9 +58,11 @@ function requestTestImage (test) {
   return Jimp
     .read(baselineImagePath)
     .then(baselineImage => {
+      console.log('baselineImage :', baselineImage)
       return Jimp
         .read(cdnUrl + requestPath)
         .then(testImage => {
+          console.log('testImage :', testImage)
```
Comment (eduardoboucas) [inline test/acceptance/visual.js]:
```suggestion
```

File: test/acceptance/visual.js [hunk 7]
```diff
@@ -59,9 +58,11 @@ function requestTestImage (test) {
   return Jimp
     .read(baselineImagePath)
     .then(baselineImage => {
+      console.log('baselineImage :', baselineImage)
```
Comment (eduardoboucas) [inline test/acceptance/visual.js]:
```suggestion
```

**Pull request merged**